src/app/qgisapp.cpp

```diff
@@ -4129,7 +4138,15 @@ void QgisApp::askUserForGDALSublayers( QgsRasterLayer *layer )
       QgsRasterLayer *rlayer = new QgsRasterLayer( sublayers[i], name );
       if ( rlayer && rlayer->isValid() )
       {
-        addRasterLayer( rlayer );
+        if ( addToGroup )
+        {
+          QgsProject::instance()->addMapLayer( rlayer, ! addToGroup );
```

Comment:
The following should be a bit easier to read.

    QgsProject::instance()->addMapLayer( rlayer, false );